src/components/coverage-statement-form/coverage-statement-form.js

```diff
@@ -0,0 +1,166 @@
+    if (this.props.onEdit) {
+      this.props.onEdit(isEditing);
+    } else {
+      this.setState({ isEditing });
+    }
+  }
+
+  handleEdit = (e) => {
+    e.preventDefault();
+    this.toggleEditing(true);
+  }
+
+  handleCancel = (e) => {
+    e.preventDefault();
+    this.toggleEditing(false);
+    this.props.initialize(this.props.initialValues);
+  }
+
+  render() {
+    let {
+      pristine,
+      isPending,
+      isEditable = this.state.isEditing,
+      // change,
+      handleSubmit,
+      onSubmit
+    } = this.props;
+    const {
+      coverageStatement
+    } = this.props.initialValues;
+    let contents;
+
+    if (isEditable) {
+      contents = (
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <div
+            data-test-eholdings-customer-resource-coverage-statement-form
+            className={styles['coverage-statement-form-editing']}
+          >
+            <CoverageStatementFields />
+            <div className={styles['coverage-statement-action-buttons']}>
+              <div
+                data-test-eholdings-customer-resource-cancel-coverage-statement-button
+                className={styles['coverage-statement-action-button']}
+              >
+                <Button
+                  disabled={isPending}
+                  type="button"
+                  role="button"
+                  onClick={this.handleCancel}
+                  marginBottom0 // gag
+                >
+                  Cancel
+                </Button>
+              </div>
+              <div
+                data-test-eholdings-customer-resource-save-coverage-statement-button
+                className={styles['coverage-statement-action-button']}
+              >
+                <Button
+                  disabled={pristine || isPending}
+                  type="submit"
+                  role="button"
+                  buttonStyle="primary"
+                  marginBottom0 // gag
+                >
+                  {isPending ? 'Saving' : 'Save'}
+                </Button>
+              </div>
+              {isPending && (
+                <Icon icon="spinner-ellipsis" />
+              )}
+            </div>
+          </div>
+        </form>
+      );
+    } else if (coverageStatement) {
+      contents = (
+        <div className={styles['coverage-statement-display']}>
+          <span data-test-eholdings-customer-resource-coverage-statement-display>
+            {coverageStatement}
+          </span>
+          <div data-test-eholdings-customer-resource-edit-coverage-statement-button>
+            <IconButton icon="edit" onClick={this.handleEdit} />
+          </div>
+        </div>
+      );
+    } else {
+      contents = (
+        <div data-test-eholdings-customer-resource-add-coverage-statement-button>
+          <Button
+            type="button"
+            onClick={this.handleEdit}
+          >
+            Set coverage statement
+          </Button>
+        </div>
+      );
+    }
```

Comment:
Would it be possible to pull this out into methods that return what they're doing? For example:

(ignore the formatting) 
```js
renderCoverageStatement() {
  return (
     <div className={styles['coverage-statement-display']}>
          <span data-test-eholdings-customer-resource-coverage-statement-display>
            {coverageStatement}
          </span>
          <div data-test-eholdings-customer-resource-edit-coverage-statement-button>
            <IconButton icon="edit" onClick={this.handleEdit} />
          </div>
     </div>
  );
}

renderEditingForm() {
  return (
    <form onSubmit={handleSubmit(onSubmit)}>
          <div
            data-test-eholdings-customer-resource-coverage-statement-form
            className={styles['coverage-statement-form-editing']}
          >
            <CoverageStatementFields />
            <div className={styles['coverage-statement-action-buttons']}>
              <div
                data-test-eholdings-customer-resource-cancel-coverage-statement-button
                className={styles['coverage-statement-action-button']}
              >
                <Button
                  disabled={isPending}
                  type="button"
                  role="button"
                  onClick={this.handleCancel}
                  marginBottom0 // gag
                >
                  Cancel
                </Button>
              </div>
              <div
                data-test-eholdings-customer-resource-save-coverage-statement-button
                className={styles['coverage-statement-action-button']}
              >
                <Button
                  disabled={pristine || isPending}
                  type="submit"
                  role="button"
                  buttonStyle="primary"
                  marginBottom0 // gag
                >
                  {isPending ? 'Saving' : 'Save'}
                </Button>
              </div>
              {isPending && (
                <Icon icon="spinner-ellipsis" />
              )}
            </div>
          </div>
        </form>
  );
}

renderSetCoverage() {
  return (
      <div data-test-eholdings-customer-resource-add-coverage-statement-button>
          <Button
            type="button"
            onClick={this.handleEdit}
          >
            Set coverage statement
          </Button>
        </div>
  );
}

render() {
    let {
      pristine,
      isPending,
      isEditable = this.state.isEditing,
      // change,
      handleSubmit,
      onSubmit
    } = this.props;
    const {
      coverageStatement
    } = this.props.initialValues;
    let contents;

    if (isEditable) {
      contents = this.renderEditingForm();
    } else if (coverageStatement) {
      contents = renderCoverageStatement();
    } else {
      contents = this.renderSetCoverage();
    }

    return (
      <div
        data-test-eholdings-embargo-form
        className={cx(styles['coverage-statement-form'], {
          'is-editing': isEditable
        })}
      >
        {contents}
      </div>
    );
  }
}
```

That way it cleans up the render method. How do people feel about that pattern? It's usually what I do when I feel like `render` is getting out of hand.

Reply:
Good idea. Done. I look forward to producing an `InlineForm` component that will make this pattern even easier to use.